lib/install.js

```diff
@@ -259,6 +259,10 @@ Installer.prototype.run = function (cb) {
     [this, this.debugActions, 'decomposeActions', 'todo'])
   if (!this.dryrun) {
     installSteps.push(
+      [this.newTracker(log, 'runPreinstallTopLevelLifecycles', 2)],
```

Comment:
Specifically: We don't need a new type of tracker, we can use runTopLevelLifecycles across both pre and postinstall lifecycles. Multiple completion trackers can be active simultaneously.